src/neo4j/mod.rs

```diff
@@ -0,0 +1,468 @@
+    /// [`Self::with_password`].
+    pub fn without_authentication(mut self) -> Self {
+        self.user = None;
+        self.pass = None;
+        self
+    }
+
+    /// Add Neo4j lab plugins to get started with the database.
+    #[must_use]
+    pub fn with_neo4j_labs_plugin(mut self, plugins: &[Neo4jLabsPlugin]) -> Self {
+        self.plugins.extend(plugins.iter().cloned());
+        self
+    }
+}
+
+type Value = Cow<'static, str>;
+
+impl Default for Neo4j {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// The actual Neo4j testcontainers image type which is returned by `container.image()`
+pub struct Neo4jImage {
+    version: String,
+    auth: Option<(String, String)>,
+    env_vars: HashMap<String, String>,
+    state: RefCell<Option<ContainerState>>,
+}
+
+impl Neo4jImage {
+    /// Return the version of the Neo4j image.
+    #[must_use]
+    pub fn version(&self) -> &str {
+        &self.version
+    }
+
+    /// Return the user/password authentication tuple of the Neo4j server.
+    /// If no authentication is set, `None` is returned.
+    #[must_use]
+    pub fn auth(&self) -> Option<(&str, &str)> {
+        self.auth
+            .as_ref()
+            .map(|(user, pass)| (user.as_str(), pass.as_str()))
+    }
+
+    /// Return the user of the Neo4j server.
+    /// If no authentication is set, `None` is returned.
+    #[must_use]
+    pub fn user(&self) -> Option<&str> {
+        self.auth().map(|(user, _)| user)
+    }
+
+    /// Return the password of the Neo4j server.
+    /// If no authentication is set, `None` is returned.
+    #[must_use]
+    pub fn password(&self) -> Option<&str> {
+        self.auth().map(|(_, pass)| pass)
+    }
+
+    /// Return the port to connect to the Neo4j server via Bolt over IPv4.
+    pub fn bolt_port_ipv4(&self) -> u16 {
+        self.state
+            .borrow()
+            .as_ref()
+            .expect("Container must be started before port can be retrieved")
+            .host_port_ipv4(7687)
+    }
+
+    /// Return the port to connect to the Neo4j server via Bolt over IPv6.
+    pub fn bolt_uri_ipv6(&self) -> u16 {
+        self.state
+            .borrow()
+            .as_ref()
+            .expect("Container must be started before port can be retrieved")
+            .host_port_ipv6(7687)
+    }
+
+    /// Return the port to connect to the Neo4j server via HTTP over IPv4.
+    pub fn http_port_ipv4(&self) -> u16 {
+        self.state
+            .borrow()
+            .as_ref()
+            .expect("Container must be started before port can be retrieved")
+            .host_port_ipv4(7474)
+    }
+
+    /// Return the port to connect to the Neo4j server via HTTP over IPv6.
+    pub fn http_uri_ipv6(&self) -> u16 {
+        self.state
+            .borrow()
+            .as_ref()
+            .expect("Container must be started before port can be retrieved")
+            .host_port_ipv6(7474)
+    }
+}
```

Comment:
I guess there is no `uri` methods anymore, so should we cleanup naming? 

Btw, is it possible to pass ENV parameter to change the ports? In that case it's also wrong to assume the ports always correct

If these ports can be changed by ENV variable passed to container, we should either:
- get ENV var to get correct port of container, otherwise do fallback to default
- remove these methods (but I like the idea to have methods to get specific ports without explicit mapping)

Reply:
> I guess there is no `uri` methods anymore, so should we cleanup naming?
 

Oh no, I though I renamed all the URI references 🙈. Will update the names.

> Btw, is it possible to pass ENV parameter to change the ports? In that case it's also wrong to assume the ports always correct

Well, kinda, maybe…
On the one hand, you can configure any possible configuration of Neo4j by setting a certain ENV. On the other hand, you have to do that through `RunnableImage`, the plain `Neo4j(Image)` type does not allow you to change the port, no ENVs are picked up implicitly. I guess you also have to change the exposed ports since the Neo4j Docker image only exposes the default ports and you also cannot do that through the module type offered here. At this point, I'd say there is no accidental part left and you'd have to be trying quite hard to change the ports. You gonna have to familiar enough with testcontainers as well as Neo4j, that I would expect that you'd then also be able to figure out how the query for the mapped ports. So I guess, within reasonable effort, you cannot change the ports, though, _technically_ you can.

Reply:
Ok, makes sense for now, we always can do a maintenance if anybody would ask

In general, even if you can override the port, it will be possible to `runnable_image..host_port_ipv6()` instead of `runnable_image.inner().bolt_port_ipv4()`. So these methods looks like a good defaults.

Thanks!